src/transformers/models/grounding_dino/modeling_grounding_dino.py

```diff
@@ -0,0 +1,3670 @@
+        # doesn't support dictionary with non-homogeneous values, such
+        # as a dict having both a Tensor and a list.
+        return [{"logits": a, "pred_boxes": b} for a, b in zip(outputs_class[:-1], outputs_coord[:-1])]
+
+    @add_start_docstrings_to_model_forward(GROUNDING_DINO_INPUTS_DOCSTRING)
+    @replace_return_docstrings(output_type=GroundingDINOObjectDetectionOutput, config_class=_CONFIG_FOR_DOC)
+    def forward(
+        self,
+        pixel_values: torch.FloatTensor,
+        input_ids: torch.LongTensor,
+        attention_mask: torch.LongTensor = None,
+        token_type_ids: torch.LongTensor = None,
+        pixel_mask: Optional[torch.BoolTensor] = None,
+        encoder_outputs: Optional[Union[GroundingDINOEncoderOutput, Tuple]] = None,
+        labels: List[Dict[str, Union[torch.LongTensor, torch.FloatTensor]]] = None,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+        return_dict: Optional[bool] = None,
+    ):
+        r"""
+        labels (`List[Dict]` of len `(batch_size,)`, *optional*):
+            Labels for computing the bipartite matching loss. List of dicts, each dictionary containing at least the
+            following 2 keys: 'class_labels' and 'boxes' (the class labels and bounding boxes of an image in the batch
+            respectively). The class labels themselves should be a `torch.LongTensor` of len `(number of bounding boxes
+            in the image,)` and the boxes a `torch.FloatTensor` of shape `(number of bounding boxes in the image, 4)`.
+
+        Returns:
+
+        Examples:
+
+        ```python
+        >>> from transformers import AutoProcessor, GroundingDINOForObjectDetection
+        >>> from PIL import Image
+        >>> import requests
+
+        >>> url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+        >>> image = Image.open(requests.get(url, stream=True).raw)
+        >>> text = "a cat."
+
+        >>> processor = AutoProcessor.from_pretrained("idea-research/grounding-dino-tiny")
+        >>> model = GroundingDINOForObjectDetection.from_pretrained("idea-research/grounding-dino-tiny")
+
+        >>> inputs = processor(images=image, text=text, return_tensors="pt")
+        >>> outputs = model(**inputs)
+
+        >>> # convert outputs (bounding boxes and class logits) to COCO API
+        >>> target_sizes = torch.tensor([image.size[::-1]])
+        >>> results = processor.image_processor.post_process_object_detection(
+        ...     outputs, threshold=0.35, target_sizes=target_sizes
+        ... )[0]
+        >>> for score, label, box in zip(results["scores"], results["labels"], results["boxes"]):
+        ...     box = [round(i, 2) for i in box.tolist()]
+        ...     print(f"Detected {label.item()} with confidence " f"{round(score.item(), 3)} at location {box}")
+        Detected 1 with confidence 0.453 at location [344.82, 23.18, 637.4, 373.83]
```

Comment:
Shouldn't it be detected "cat"?

Reply:
Since `GroundingDINO` logits have shape `batch size, num_queries, hidden_state` the authors extract the labels from the input text using a `text_threshold` see [here](https://github.com/IDEA-Research/GroundingDINO/blob/main/groundingdino/util/inference.py#L92) I wasn't sure where to add this step, since it requires the `input_ids` and the `tokenizer` I thought that we could add a method to `GroundingDINOProcessor` that would do that, however, I don't believe this would follow HF standards

Reply:
Usually classes are predefined and mapped using the `id2label` dictionary, but this seems a very particular case. 
Leaving the label as the "id" seems not very helpful for the user. Curious to know Niels opinion on this :)

Reply:
@amyeroberts  for `post_process_object_detection` the labels represent the indices from `torch.max(..., dim=-1)` the logits have shape `batch_size, num_queries, embed_dim`, and unfortunately to get the text associated with each bbox we also need the `input_ids` and the `tokenizer`, thus the  `post_process_grounded_object_detection` in `GroundingDinoProcessor`  as a `GroundingDino` specific post-processing. This way we keep the `post_process_object_detection` signature and standard